### PR TITLE
Added test case to validate delete event

### LINKED ISSUE #354: Add test coverage for delete events in e2e tests
In e2e test suite, we mock resource creation with fake client and verify the BotKube event generations.
https://github.com/infracloudio/botkube/blob/develop/test/e2e/notifier/create/create.go#L46

We need to add similar test cases to verify delete event handling.
We can use [Delete](https://godoc.org/k8s.io/client-go/kubernetes/typed/core/v1#PodInterface) API for Deleting a resource.

Add test cases to validate the following behaviour:

- [ ] Validate delete events are handled for a resource with `events: [all]` in test config https://github.com/infracloudio/botkube/blob/develop/test/resource_config.yaml - @ShubhamSood1406

- [x] Validate delete events are handled with only `events: [delete]` config for a resource - @Divya063

- [x] Validate delete events are skipped when `delete` event is not configured for a resources in test config. e.g `events: [create, update, error]`- @A-kanksh-a 

- [x] Validate that delete events are skipped when an event occurs for a namespace which is not configured - @A-kanksh-a 

- [x] Validate that delete events are skipped when an event occurs for a resource which is not added in resource_config - @A-kanksh-a 

Each test case can be added as a separate PR. Multiple PR as expected for this task. 
Guidelines for contributors:
- Post comment on the issue informing others about the test case you want to add
- Mark the issue as done once the PR is merged
- Please make sure that no one else is working on the same test case to avoid duplicate PRs

### PR DESCRIPTION
**Pull request by Divya063**
Adds test case - Validate delete events are handled with only events: [delete] config for a resource.
Fixes part of #354

### CODE REVIEW

File: test/e2e/notifier/delete/delete.go [hunk 1]
```diff
@@ -51,9 +54,81 @@ func (c *context) testSKipDeleteEvent(t *testing.T) {
...
+func (c *context) testDeleteEvent(t *testing.T) {
+	events := []config.EventType{"update", "error", "create"}
+
+	// Modifying AllowedEventKindsMap to remove events other than delete for pod resource
+	for _, event := range events {
+		delete(utils.AllowedEventKindsMap, utils.EventKind{Resource: "v1/pods", Namespace: "all", EventType: event})
+	}
+	// test scenarios
+	tests := map[string]testutils.DeleteObjects{
+		"perform delete operation with only delete event": {
```
Comment (PrasadG193) [inline test/e2e/notifier/delete/delete.go:66]:
```suggestion
		"perform delete operation and configure BotKube to listen only delete events": {
```

**Comment by PrasadG193** [comment]
Thanks, @Divya063 for the PR :pray: 
I have a minor suggestion, rest looks good

**Comment by Divya063** [comment]
@PrasadG193 made the changes.

**Review by PrasadG193 — APPROVED** [review]

**Comment by chetanpdeshmukh** [comment]
Hello @Divya063 ,
Thanks again for the PR!

Thank you for contributing to BotKube. Could you please fill out [this form](https://docs.google.com/forms/d/e/1FAIpQLSdULT22oZlpHPMXKVEtV4fZ6R8vdGDC2iFvMhoG6IhZkK1SFA/viewform?usp=sf_link), so we can send the well deserved awesome swags :slightly_smiling_face:

Team BotKube

**Pull request merged**